pkg/controller/elasticsearch/sset/list.go

```diff
@@ -54,7 +54,12 @@ func (l StatefulSetList) ObjectMetas() []metav1.ObjectMeta {
 func (l StatefulSetList) ToUpdate() StatefulSetList {
 	toUpdate := StatefulSetList{}
 	for _, s := range l {
-		if s.Status.UpdateRevision != "" && (s.Status.UpdateRevision != s.Status.CurrentRevision) {
+		// When using on delete current revision is never reset to update revision
+		// just looking that the update/current revision therefore does not work when reverting
+		// to previous revision and gives constant false positives after an initial update.
```

Comment:
This comment block looks a bit like it's commenting some removed code. Maybe it's enough to just change the function comment to `ToUpdate filters the StatefulSetList to the ones having some replicas not updated yet`.

Reply:
Yes it does. I wanted to capture the fact that the behaviour of stateful sets differs significantly depending on which strategy you use.